src/Compiler/SyntaxTree/ParseHelpers.fs

```diff
@@ -1127,10 +1152,12 @@ let mkLetExpression
 
             SynExpr.LetOrUse(
                 isRec,
-                isUse, // Pass through the isUse flag from binding info
+                isUse,
+                true,
+                false,
```

Comment:
Are all `2^4` possibilities valid?
If not, can we do better?


I do not want to dogmatically oppose booleans, but I think readability suffered a little especially for literal `true`/`false` values passed in without naming the argument.

Reply:
I guess as we use something like 

```fsharp
type BindingKind = 
    | Let                    // let x = ...
    | LetRec                 // let rec f = ...
    | Use                    // use x = ...
    | LetBang                // let! x = ...
    | UseBang                // use! x = ...

type BindingSource =
    | UserWritten            // From source code
    | CompilerGenerated      // Generated during compilation

type SynExpr =
    | LetOrUse of
        kind: BindingKind *
        source: BindingSource *
        bindings: SynBinding list *
        body: SynExpr *
        range: range *
        trivia: SynExprLetOrUseTrivia
```

Reply:
And each of the bindingkinds can be compiler generated? I thought only vanilla `let` and `let!` are

Reply:
No only let! as the initial `isFromSource` was only available in `LetOrUseBang`. So it will be some something like

```fsharp
type BindingSource =
    | UserWritten            // From source code
    | CompilerGenerated 

type BindingKind = 
    | Let.                   // let x = ...
    | LetRec              // let rec f = ...
    | Use                    // use x = ...
    | LetBang of source: BindingSource // or we can still use a bool here :) 
    | UseBang                // use! x = ...

type SynExpr =
    | LetOrUse of
        kind: BindingKind *
        bindings: SynBinding list *
        body: SynExpr *
        range: range *
        trivia: SynExprLetOrUseTrivia
```